### PR TITLE
kuviz has unique name 

### PR DESCRIPTION
**Pull request by oleurud**
Solves a part of: https://github.com/CartoDB/cartoframes/issues/1451

### CODE REVIEW

File: spec/requests/carto/api/custom_visualizations_controller_spec.rb [hunk 1]
```diff
@@ -337,6 +394,38 @@
...
+    end
+
+    it 'works if if_exists is fail and name does not exists' do
+      put_json api_v4_kuviz_update_viz_url(api_key: @user.api_key, id: @kuviz.id), name: 'other name', if_exists: 'fail' do |response|
+        expect(response.status).to eq(200)
+        expect(response.body[:url].present?).to be true
+      end
+    end
+
+    it 'works if if_exists is replace and name already exists' do
```
Comment (gonzaloriestra) [inline spec/requests/carto/api/custom_visualizations_controller_spec.rb]: What about testing also that the file is overridden in this case?

**Comment by oleurud** [comment]
After working in the client, I realized I must change the default behavior:
- create: if exists a kuviz with the same name: fail
- update: if exists a kuviz with the same name: replace

File: spec/requests/carto/api/custom_visualizations_controller_spec.rb (hunk 1 above)
Comment (oleurud) [reply spec/requests/carto/api/custom_visualizations_controller_spec.rb]: If I am not wrong, nothing changes in the asset and the associated file. What do you mean?

Comment (gonzaloriestra) [reply spec/requests/carto/api/custom_visualizations_controller_spec.rb]: Sorry, I meant the visualization. To check that if you change the name, it gets updated. We are only testing the response of the methods, but we could check that it also does what we expect (removing duplicates, updating attributes, etc.).

**Comment by oleurud** [comment]
I added the `not merge yet` because we need to merge things in `CARTOframes` and `carto-python` before this one

**Comment by oleurud** [comment]
Sorry @gonzaloriestra we have a last minute requirement change https://github.com/CartoDB/cartoframes/issues/1451#issuecomment-574622505 can you please check the last commit?

**Comment by gonzaloriestra** [comment]
LGTM :+1:

**Pull request merged**